client/signup/steps/domains/index.jsx

```diff
@@ -593,6 +593,40 @@ export class RenderDomainsStep extends Component {
 		await this.props.shoppingCartManager.addProductsToCart( productsToAdd ).then( () => {
 			this.setState( { isCartPendingUpdateDomain: null } );
 		} );
+
+		// Sort products to ensure the user gets the best deal with the free domain bundle promotion.
+		this.sortProductsByPriceDescending();
+	}
+
+	async sortProductsByPriceDescending() {
+		// Get products from cart.
+		const productsInCart = this.props.cart.products;
+
+		// Sort products by price descending, considering promotions.
```

Comment:
Sort the domains descending by `item_subtotal_integer`. However, the most expensive domain already in the cart has an `item_subtotal_integer = 0` because the bundle promotion has already been applied. So, if `item_subtotal_integer == 0`, we use the lowest non-zero `new_price` in `cost_overrides` (multiplied by 100). This is the price after all promotions have been applied. If there are no non-zero `new_prices`, we assume the only promotion being applied is the free domain with bundle, so we use `item_original_cost_integer`.